zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -27,9 +32,30 @@ pub struct Chain {
     sapling_nullifiers: HashSet<sapling::Nullifier>,
     orchard_nullifiers: HashSet<orchard::Nullifier>,
     partial_cumulative_work: PartialCumulativeWork,
+    pub(crate) history_tree: HistoryTree,
 }
 
 impl Chain {
+    /// Create a new empty non-finalized chain with the given history tree.
+    ///
+    /// The history tree must contain the history of the previous (finalized) blocks.
+    pub fn new(history_tree: HistoryTree) -> Self {
+        Chain {
+            blocks: Default::default(),
+            height_by_hash: Default::default(),
+            tx_by_hash: Default::default(),
+            created_utxos: Default::default(),
+            spent_utxos: Default::default(),
+            sprout_anchors: Default::default(),
+            sapling_anchors: Default::default(),
+            sprout_nullifiers: Default::default(),
+            sapling_nullifiers: Default::default(),
+            orchard_nullifiers: Default::default(),
+            partial_cumulative_work: Default::default(),
+            history_tree,
```

Comment:
If  everything else in this `Chain` instance is empty/default, can this instance of `HistoryTree` be empty/default?

Reply:
>      /// The history tree must contain the history of the previous (finalized) blocks.

Reply:
Yep, the history tree will never be empty. 
(When there is a network upgrade it switches from a full tree to a single-node tree. BTW this is something that must be handled after this PR)